connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -409,6 +412,13 @@ public String groupId() {
         return null;
     }
 
+    public String kafkaClusterId() {
+        if (kafkaClusterId == null) {
+            kafkaClusterId = ConnectUtils.lookupKafkaClusterId(this);
+        }
+        return kafkaClusterId;
+    }
```

Comment:
Should we move the `lookupKafkaClusterId` logic from `ConnectUtils` to this class, so that nobody accidentally starts using the variant that creates and discards an admin client once per call?

Reply:
That's a good idea. I've move `lookupKafkaClusterId` to `WorkerConfig`. We should be able to make this method not public by updating some of the tests but I've not done that since some of the callers are still using Easymock/Powermock so they are being updated.

Reply:
Yeah, was wondering about the fallout on tests. This seems fine for now 👍